packages/dds/tree/src/simple-tree/toMapTree.ts

```diff
@@ -62,6 +65,32 @@ import type { IFluidHandle } from "@fluidframework/core-interfaces";
  * schema than on the input data.
  */
 
+/**
+ * Options for {@link mapTreeFromNodeData}.
+ */
+interface MapTreeFromNodeDataOptions {
+	/**
+	 * An optional context which, if present, will allow defaults to be created by {@link ContextualFieldProvider}s.
+	 * If absent, only defaults from {@link ConstantFieldProvider}s will be created.
+	 */
+	context?: NodeIdentifierManager;
+	/**
+	 * The stored schema and policy to be used for validation, if the policy says schema
+	 * validation should happen. If it does, the input tree will be validated against this schema + policy, and an error will
+	 * be thrown if the tree does not conform to the schema. If undefined, no validation against the stored schema is done.
+	 *
+	 * TODO:BUG: AB#9131
+	 * This schema validation is done before defaults are provided.
+	 * This can not easily be fixed by reordering things within this implementation since even at the end of this function defaults requiring a context may not have been filled.
+	 * This means schema validation reject required fields getting their value from a default like identifier fields.
+	 */
+	schemaValidationPolicy?: SchemaAndPolicy;
+	/**
+	 * A flag that determines if an enablable allowed type is permitted in the node data, defaults to false.
+	 */
+	allowNonEnabledTypes?: boolean;
```

Comment:
I think it's important that that you don't have to enable all types at once.

If I have two different teams working on the same application, and both are working on features that use an enablable type, they should be able to enable them one at a time, and not have to wait for the feature to ship to start the second one or do them both at exactly the same time.

This I think you need to either provide a schema which reflects whats actually allowed in the document (the stored schema that's part of the optional schema nd policy above does that btw, but I'm going to be removing that from toMap tree most likley) OR provide a set of enablements (and which on must be able to be referred to somehow, which we need anyway to be able to do schema upgrades one at a time anyway).

Reply:
This flag is just an implementation detail of the `mapTreeFromNodeData` funtion (which correct me if I'm wrong, I don't think is public) and is used to control when encountering an enablable causes a runtime error so that we can not error when loading the data.

Reply:
But if we have a document where one type has been enabled, and another has not yet, how do we error if there is an instance of the no enabled one and not error if there is an instance of the enabled one?

Of do we just always error, in which case why is there a flag?

Reply:
> don't error when we see an enablable in view.initialize

This seems like a case that should get a test in src/test/shared-tree/schematizingTreeView.spec.ts

Reply:
Ah, good point, I had put it in `sharedTree.spec.ts` but def should have one at the schematizingTreeView layer as well.